controllers/usersignup/usersignup_controller.go

```diff
@@ -112,6 +118,51 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, nil
 	}
 
+	// REMOVE THIS BLOCK AFTER MIGRATION - See additional comments further below in this function
+	originalUserSignupName, ok := userSignup.Annotations[migrationAnnotationName]
+	if ok {
+		// If the annotation has been set, then it indicates that this UserSignup has been migrated, and that the
+		// *original* UserSignup should be deleted.  The value of the annotation is the name of the original UserSignup
+		// resource that should now be deleted.
+		//
+		// This block of migration code should:
+		//
+		// 1) Delete the original UserSignup, then
+		// 2) Remove the migration-replaces annotation
+		userSignupToDelete := &toolchainv1alpha1.UserSignup{}
+		err = r.Client.Get(context.TODO(), types.NamespacedName{
+			Namespace: request.Namespace,
+			Name:      originalUserSignupName,
+		}, userSignupToDelete)
+		if err != nil {
+			// If the annotation exists however the original UserSignup isn't found, then remove the annotation
+			if errors.IsNotFound(err) {
+				// Remove the annotation
+				delete(userSignup.Annotations, migrationAnnotationName)
+				// Update the UserSignup
+				err = r.Client.Update(ctx, userSignup)
+				if err != nil {
+					return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup,
+						r.setStatusMigrationFailedCleanup, err, fmt.Sprintf("Failed to remove migration annotation"))
+				}
+			} else {
+				return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup,
+					r.setStatusMigrationFailedLookup, err, fmt.Sprintf("Failed to lookup original UserSignup [%s]",
+						userSignup.Annotations[migrationAnnotationName]))
+			}
+		}
+
+		// Delete the original UserSignup
+		err = r.Client.Delete(context.TODO(), userSignupToDelete)
+		if err != nil {
+			return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup, r.setStatusMigrationFailedCleanup,
+				err, fmt.Sprintf("Failed to remove original UserSignup [%s]",
+					userSignup.Annotations[migrationAnnotationName]))
+		}
+		// Requeue so that the annotation will now be removed also
+		return reconcile.Result{Requeue: true}, nil
```

Comment:
I wonder if we want to add some delay (less than a second?) here...

Reply:
I think a 1 second delay is acceptable, added.